### ISSUE TITLE
Support for UUID in BSON

### ISSUE THREAD

**Issue by wardlem** [felipenoris/Mongoc.jl#75]
Greetings,

I am working on a project that requires storing binary data and UUIDs in Mongo.  I see no obvious way of doing this with the library as it is (or at least no documented way) except to convert the binary data to EJSON and then evaluate it.  This is rather awkward.  It would be nice for this to work:

```julia
BSON(
    "uuid" => UUIDs.uuid4(), 
    "bindata" => [0x01, 0x02, 0x03, 0x04],
)

# Equivalent to

BSON("""{
    "uuid": {"\$binary": {
        "base64": "$(base64encode(UUIDs.uuid4().value))",
        "subType": "$(convert(UInt8, Mongoc.BSON_SUBTYPE_UUID))"
    }},
    "bindata": {"\$binary": {
        "base64": "$(base64encode([0x01, 0x02, 0x03, 0x04]))",
        "subType": "$(convert(UInt8, Mongoc.BSON_SUBTYPE_BINARY))"
    }}
}""")
```

I am more than willing to implement this if needed.

**Comment by wardlem** [comment]
You know what, I am completely wrong about the binary data.  It does work as expected when passing in a `Vector{UInt8}`.  I must have been testing it wrong.  It would still be nice to have built-in support for UUIDs.

**Comment by wardlem** [comment]
Created #76 for this.

**Comment by felipenoris** [comment]
Closed by df944474e259377c04bd4e82da0c617d36fa13ef
